src/Form/FormAutosuggest.jsx

```diff
@@ -334,9 +380,23 @@ FormAutosuggest.propTypes = {
   /** Specifies the placeholder text for the input. */
   placeholder: PropTypes.string,
   /** Specifies values for the input. */
-  value: PropTypes.string,
-  /** Informs user has errors. */
-  errorMessageText: PropTypes.string,
+  value: PropTypes.shape({
+    userProvidedText: PropTypes.string,
+    selectionValue: PropTypes.string,
+    selectionId: PropTypes.string,
+  }),
+  /** Specifies if empty values trigger an error state */
+  valueRequired: PropTypes.bool,
+  /** Informs user they must input a value. */
+  valueRequiredErrorMessageText: PropTypes.string,
```

Comment:
I assume that there should be `valueRequiredErrorMessageText` provided if `valueRequired` equals `true`. Please, consider:

```
import { requiredWhen } from '../utils/propTypes';

...

valueRequiredErrorMessageText: requiredWhen(PropTypes.string, 'valueRequired'),
```

And the same for `selectionRequired` and `selectionRequiredErrorMessageText`, `customError` and `customErrorMessageText`.

Another option, if no message is provided and there should only be a red box to indicate the error, I would remove the `x` under the input, otherwise it will appear incomplete.![image](https://github.com/openedx/paragon/assets/36768631/eeca2480-32d2-479a-bdf1-56d51b998c40)

Reply:
@monteri I have updated these to use `requiredWhen`

https://github.com/openedx/paragon/blob/f8a4e5ee118199936c3ff6653e3dfb389d12fffa/src/Form/FormAutosuggest.jsx#L421-L430